pkg/util/helper/workstatus.go

```diff
@@ -73,41 +73,28 @@ func AggregateResourceBindingWorkStatus(
 
 	fullyAppliedCondition := generateFullyAppliedCondition(binding.Spec, aggregatedStatuses)
 
-	err = retry.RetryOnConflict(retry.DefaultRetry, func() (err error) {
-		currentBindingStatus := binding.Status.DeepCopy()
-
-		binding.Status.AggregatedStatus = aggregatedStatuses
-		// set binding status with the newest condition
-		meta.SetStatusCondition(&binding.Status.Conditions, fullyAppliedCondition)
-		if reflect.DeepEqual(binding.Status, *currentBindingStatus) {
-			klog.V(4).Infof("New aggregatedStatuses are equal with old resourceBinding(%s/%s) AggregatedStatus, no update required.",
-				binding.Namespace, binding.Name)
-			return nil
-		}
-
-		updateErr := c.Status().Update(context.TODO(), binding)
-		if updateErr == nil {
+	var operationResult controllerutil.OperationResult
+	if err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		operationResult, err = UpdateStatus(context.Background(), c, binding, func() error {
+			binding.Status.AggregatedStatus = aggregatedStatuses
+			// set binding status with the newest condition
+			meta.SetStatusCondition(&binding.Status.Conditions, fullyAppliedCondition)
 			return nil
-		}
-
-		updated := &workv1alpha2.ResourceBinding{}
-		if err = c.Get(context.TODO(), client.ObjectKey{Namespace: binding.Namespace, Name: binding.Name}, updated); err == nil {
```

Comment:
The main problem is that the update will not be performed immediately after obtaining the latest binding, which  subject to `retry.RetryOnConflict` back-off.

Reply:
I get it. should be `get` --> `update` --> `back-off`, instead of `get` --> `back-off` --> `update`.
Conflict might always happen.